File: 프로그래머스/1/42862. 체육복/체육복.py
def solution(n, lost, reserve):
    lost, reserve = set(lost), set(reserve)
    reserve_s, lost_s = set(reserve) - set(lost), set(lost) - set(reserve)

    for extra in sorted(reserve_s):
        if extra-1 in lost_s:
           lost_s.remove(extra-1)
        elif extra+1 in lost_s:
           lost_s.remove(extra+1)

    return n - len(lost_s)

File: 프로그래머스/1/42862. 체육복/체육복.js
function solution(n, lost, reserve) {
    const notDupLost = lost.filter(x => !reserve.includes(x));
    const notDupReserve = reserve.filter(x => !lost.includes(x));
    notDupLost.sort((a, b) => a - b);
    notDupReserve.sort((a, b) => a - b);
    
    for(let extra of notDupReserve){
        const left = notDupLost.indexOf(extra-1);
        const right = notDupLost.indexOf(extra+1);
        
        if(left !== -1){
            notDupLost.splice(left, 1);
        } else if(right !== -1){
            notDupLost.splice(right, 1);
        }
    }
    
    return(n - notDupLost.length);
}